<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>dbe::BatchChangeWidget</class>
 <widget class="QWidget" name="dbe::BatchChangeWidget">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>611</width>
    <height>974</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="toolTip">
   <string/>
  </property>
  <property name="whatsThis">
   <string/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_6">
   <item>
    <widget class="QLabel" name="TitleLabel">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>40</height>
      </size>
     </property>
     <property name="text">
      <string>Batch Change of multiple objects</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="FirstStepLabel">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>20</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>16777215</width>
       <height>16777209</height>
      </size>
     </property>
     <property name="text">
      <string>First Step: Select the object you want to edit</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="Label1">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>15</height>
      </size>
     </property>
     <property name="text">
      <string>You can set multiple filters, Only the objects matching all of them will be selected.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="Label2">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>15</height>
      </size>
     </property>
     <property name="text">
      <string>You can see the matching selected objects clicking on the &quot;Show matching objects&quot; button here below.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="FilterGroupBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Filter</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="ClassLabel">
          <property name="text">
           <string>Class:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="ClassCombo">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>20</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Select a Class</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="SubclassesCheckBox">
        <property name="text">
         <string>Check if you want the changes to take effect in the subclasses as well</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="LabelUid">
          <property name="text">
           <string>UID:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="FilterUid">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>20</height>
           </size>
          </property>
          <property name="toolTip">
           <string>You can enter a string to be used as filter on the UID of the objects in the object list</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QGroupBox" name="AttributeBox">
        <property name="title">
         <string>Attributes</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QLabel" name="SelectAttributeLabel">
               <property name="toolTip">
                <string>Select an Attribute and a value to be used as filter to select objects in the object list</string>
               </property>
               <property name="text">
                <string>Select matching Attribute value</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="AttributeComboBox">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>20</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Select an Attribute and a value to be used as filter to select objects in the object list</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QLineEdit" name="AttributeValueFilter">
             <property name="minimumSize">
              <size>
               <width>0</width>
               <height>20</height>
              </size>
             </property>
             <property name="toolTip">
              <string>Select an Attribute and a value to be used as filter to select objects in the object list</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="RelationshipBox">
        <property name="title">
         <string>Relationships</string>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QLabel" name="SelectRelationshipLabel">
               <property name="toolTip">
                <string>Select a Relationship and a value to be used as filter to select objects in the object list</string>
               </property>
               <property name="text">
                <string>Select matching Relationship value</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="RelationshipComboBox">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>20</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Select a Relationship and a value to be used as filter to select objects in the object list</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QComboBox" name="RelationshipValueFilter">
             <property name="editable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="MatchButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>20</height>
         </size>
        </property>
        <property name="toolTip">
         <string>Click to see the list of objects that will be affected by the batch changes</string>
        </property>
        <property name="text">
         <string>Show matching objects</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="Line" name="Line1">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="SecondStepLabel">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>20</height>
      </size>
     </property>
     <property name="text">
      <string>Second step: Set new values for attributes and relationships</string>
     </property>
     <property name="textFormat">
      <enum>Qt::AutoText</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="NewAttributeGroupBox">
     <property name="toolTip">
      <string>Choose attribute and new value for it</string>
     </property>
     <property name="title">
      <string>New Attribute Value</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QComboBox" name="NewAttributeComboBox">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>20</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="AttributeEdit">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>20</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="ChangeAtrributeCheckBox">
     <property name="text">
      <string>Check if you want change attribute</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="NewRelationshipGroupBox">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="toolTip">
      <string>Batch change of relationships are not yet implemented.</string>
     </property>
     <property name="whatsThis">
      <string>Batch change of relationships are not yet implemented.</string>
     </property>
     <property name="title">
      <string>New RelationShip value</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_6">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QComboBox" name="NewRelationshipComboBox">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>20</height>
           </size>
          </property>
          <property name="toolTip">
           <string>Batch change of relationships are not yet implemented.</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="RelationshipEdit">
          <property name="editable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="AddCheckBox">
          <property name="text">
           <string>Check to add relationship or uncheck to substitue value</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="ChangeRelationshipCheckBox">
     <property name="text">
      <string>Check if you want change relationship</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="Line" name="Line2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="ThirdStepLabel">
     <property name="minimumSize">
      <size>
       <width>0</width>
       <height>20</height>
      </size>
     </property>
     <property name="toolTip">
      <string>Make the changes active</string>
     </property>
     <property name="text">
      <string>Third step: Make changes or cancel</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QGridLayout" name="GridLayout">
     <item row="0" column="0">
      <widget class="QPushButton" name="ChangeButton">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>20</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>120</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Click to make your changes active</string>
       </property>
       <property name="text">
        <string>Make change</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="CancelButton">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>20</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>100</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Click to cancel the changes. Or click when you are finished with the batch changes you want to perform.</string>
       </property>
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
